src/api/app/models/staging/workflow.rb

```diff
@@ -129,13 +130,16 @@ def update_managers_group
     reviewer_role = Role.find_by_title('reviewer')
     relationship = project.relationships.find_by(group: old_managers_group, role: reviewer_role)
     if project.relationships.exists?(group: new_managers_group, role: reviewer_role)
-      relationship.destroy
-    else
+      relationship.destroy if relationship.present?
```

Comment:
I don't get it. How would it not be present if it exists?

Reply:
> I don't get it. How would it not be present if it exists?

In few words: `exists?` is for `old_managers_group`, `.present?` is for `new_managers_group`.

`relationship` (assigned in line 130) is the current record from the table relationships, with the group `old_managers_group`.

The first `if` (line 131) checks for a relationship with the group `new_managers_group`. In this case, we don't have to create a new record for the `new_managers_group` group. However, we have to remove the relationship with the old group, only if that one is present.

Reply:
I followed your suggestion and modified the code. Please take a look again.